### PR TITLE
Fix problem_type to match with the applied loss function for distillbert sequence classification 

### LINKED ISSUE #12014: MIssmatch between problem_type and loss functions in DistillBert for sequence classification
## Environment info
- Platform: Windows 
- Python version: 3.8
- PyTorch version (GPU?): 1.8

### Who can help
@LysandreJik

Models:
Distillbert for sequence classification

## Information
The problem_type in config is not correct with the loss function applied. Can be seen [here](https://github.com/huggingface/transformers/blob/242ec31aa59b358e631d981b545fd08330584ea8/src/transformers/models/distilbert/modeling_distilbert.py#L649) 

Eg. for the problem type "single_label_classification", the CrossEntropyLoss is used, and the logic to detect "single_label_classification" indicates there should be multiple classes. So its just that the labels of problem_type are switched. Hence, if someone uses actually uses this config of probelm_type to choose  their task (binary, or multiclass), the applied loss would be inccorect 

## Expected behavior
For problem_type, the "single_label_classification", the BCE loss should be applied

### PR DESCRIPTION
**Pull request by sidhantls**
# What does this PR do?
The problem_type in config is not correct with the loss function applied. Can be seen [here](https://github.com/huggingface/transformers/blob/242ec31aa59b358e631d981b545fd08330584ea8/src/transformers/models/distilbert/modeling_distilbert.py#L649). 

This PR fixes this so that the applied loss is consistent with the problem type: `BCEWithLogitsLoss` is applied for the `problem_type` of `single_label_classification`, and `CrossEntropyLoss` is applied for the problem type `multi_label_classification`

Fixes #12014

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?
@LysandreJik

### CODE REVIEW

**Comment by LysandreJik** [comment]
cc @abhi1thakur @sgugger

**Comment by sgugger** [comment]
No, this is incorrect. What `single_label_classification` means each sample can only have one label (but there could be multiple classes) so the loss to use is cross entropy. `multi_label_classification` means each sample can have zero or several labels, so in this case we use bce (because there can't be a softmax).
